src/styles/utils/_loops.scss

```diff
@@ -0,0 +1,26 @@
+@for $day from 1 through 31 {
+  .p-#{$day}::before{
+    content:"#{$day}";
+  }
+}
+
+$days:
+  'mon' 0,
+  'tue' 1,
+  'wed' 2,
+  'thu' 3,
+  'fri' 4,
+  'sat' 5,
+  'sun' 6;
+
+@each $day, $index in $days {
+  .calendar--start-day--#{$day} > :first-child {
+    margin-left: #{($cell-side * $index) + ($cell-gap * (($index * 2) + 1))};
+  }
+}
+
+@for $last-day from 28 to 31 {
+  .calendar--month-length--#{$last-day} :nth-child(n + #{$last-day + 1}) {
+    display: none;
+  }
+}
```

Comment:
These @each and @for loop can go in calendar block and can be nested.

Reply:
I thought it can be handy to have them all in one place, but you also have a point. Changed.